src/python/aqueduct_executor/operators/connectors/data/s3.py

```diff
@@ -138,7 +166,7 @@ def extract(self, params: extract.S3Params) -> Any:
                 files.append(self._fetch_object(key, params))
 
             if params.artifact_type == ArtifactType.TABLE and params.merge:
-                return pd.concat(files)
+                return pd.concat(files, ignore_index=True)
```

Comment:
So it's technically a redundant call, but it's mostly here to alleviate confusion. When I was checking the behavior of merge, I was confused for a big because it appeared that merging would ignore indexes, but when I tracked down the concatenation behavior here, it was just a vanilla `pd.concat()` call. It turns out that when we serialize the data into storage, `_write_table_output` uses `output_str = cast(str, output.to_json(orient="table", date_format="iso", index=False))`, which I believe gets rid of the indexes.

So to make the code more explainable, I decided to ignore_indexes earlier on. @cw75 since you implemented merge is my interpretation correct?

Reply:
I can add a comment to explain this a little better.

Reply:
Cool that makes sense! I was asking just in case we are fixing any bug here

Reply:
@kenxu95 I remember `merge` is indeed just a `pd.concat()`, and noting more than that.

Reply:
But the result the user gets back is equivalent to `pd.concat(ignore_index=True)` - if I compare with `pd.concat()` the test case fails.

Reply:
I see. If I understand correctly, this is because although the merge itself uses `pd.concat()`, the index in the final output was gone, as it's removed during the serialization step? But if so, the problem seems independent of the `merge` step?

Reply:
It is independent yeah - I guess the question is whether we want this ignore_index behavior or not? I wasn't sure why this was here in the serialization code to begin with. Should I be fixing serialization or leaving the `pd.concat(ignore_index=True)` for better documentation.

Reply:
I also don't understand why we should ignore the index on behalf of the user...Did I write the serialization part? Seems we should fix the serialization side.

Reply:
I think I touched it last but I'm pretty sure I copied it from existing code.